setup-homebrew/main.sh

```diff
@@ -193,35 +202,37 @@ else
         cd -
     fi
 
-    if [[ -d "${HOMEBREW_CORE_REPOSITORY}" && "${UPDATE_CORE}" == "true" ]]; then
-        # Migrate linuxbrew-core to homebrew-core
-        HOMEBREW_CORE_REPOSITORY_ORIGIN="$(git -C "$HOMEBREW_CORE_REPOSITORY" remote get-url origin 2>/dev/null)"
-        if [[ "$HOMEBREW_CORE_REPOSITORY_ORIGIN" == "https://github.com/Homebrew/linuxbrew-core" ]]
-        then
-            git -C "$HOMEBREW_CORE_REPOSITORY" remote set-url origin "https://github.com/Homebrew/homebrew-core"
-        fi
+    if [[ "${UPDATE_CORE}" == "true" ]]; then
+        if [[ -d "${HOMEBREW_CORE_REPOSITORY}" ]]; then
+            # Migrate linuxbrew-core to homebrew-core
+            HOMEBREW_CORE_REPOSITORY_ORIGIN="$(git -C "$HOMEBREW_CORE_REPOSITORY" remote get-url origin 2>/dev/null)"
+            if [[ "$HOMEBREW_CORE_REPOSITORY_ORIGIN" == "https://github.com/Homebrew/linuxbrew-core" ]]; then
+                git -C "$HOMEBREW_CORE_REPOSITORY" remote set-url origin "https://github.com/Homebrew/homebrew-core"
+            fi
 
-        ohai "Fetching Homebrew/core..."
-        git_retry -C "$HOMEBREW_CORE_REPOSITORY" fetch --force origin
-        git -C "$HOMEBREW_CORE_REPOSITORY" remote set-head origin --auto
-        git -C "$HOMEBREW_CORE_REPOSITORY" checkout --force -B master origin/HEAD
-    elif [[ -n "${HOMEBREW_NO_INSTALL_FROM_API-}" ]] && [[ ! -d "${HOMEBREW_CORE_REPOSITORY}" ]]; then
-        ohai "Fetching Homebrew/core..."
-        git_retry clone https://github.com/Homebrew/homebrew-core "${HOMEBREW_CORE_REPOSITORY}"
+            ohai "Fetching Homebrew/core..."
+            git_retry -C "$HOMEBREW_CORE_REPOSITORY" fetch --force origin
+            git -C "$HOMEBREW_CORE_REPOSITORY" remote set-head origin --auto
+            git -C "$HOMEBREW_CORE_REPOSITORY" checkout --force -B master origin/HEAD
+        else
+            ohai "Cloning Homebrew/core..."
+            git_retry clone https://github.com/Homebrew/homebrew-core "${HOMEBREW_CORE_REPOSITORY}"
+        fi
     fi
 
-    if [[ "${HOMEBREW_TAP_REPOSITORY-}" != "${HOMEBREW_CASK_REPOSITORY}" ]] && [[ -d "${HOMEBREW_CASK_REPOSITORY}" && "${UPDATE_CASK}" == "true" ]]; then
-        ohai "Fetching Homebrew/cask..."
-        git_retry -C "$HOMEBREW_CASK_REPOSITORY" fetch --force origin
-        git -C "$HOMEBREW_CASK_REPOSITORY" remote set-head origin --auto
-        git -C "$HOMEBREW_CASK_REPOSITORY" checkout --force -B master origin/HEAD
-    elif [[ "$GITHUB_REPOSITORY" =~ ^.+/homebrew-cask(-.+)*$ ]] && [[ ! -d "${HOMEBREW_CASK_REPOSITORY}" ]]; then
-        ohai "Fetching Homebrew/cask..."
-        git_retry clone https://github.com/Homebrew/homebrew-cask "${HOMEBREW_CASK_REPOSITORY}"
+    if [[ "${UPDATE_CASK}" == "true" ]]; then
+        if [[ -d "${HOMEBREW_CASK_REPOSITORY}" ]]; then
```

Comment:
Might still need the `[[ "${HOMEBREW_TAP_REPOSITORY-}" != "${HOMEBREW_CASK_REPOSITORY}" ]]` or it could fetch twice given `HOMEBREW_TAP_REPOSITORY` is fetched earlier unconditionally.

Reply:
@Bo98 thought of same thing in car 🍻. Done for core and cask.